x-pack/plugin/rollup/src/internalClusterTest/java/org/elasticsearch/xpack/downsample/DownsampleClusterDisruptionIT.java

```diff
@@ -0,0 +1,442 @@
+            ensureGreen();
+
+            final String sourceIndex = randomAlphaOfLength(10).toLowerCase(Locale.ROOT);
+            final String rollupIndex = randomAlphaOfLength(11).toLowerCase(Locale.ROOT);
+            long startTime = LocalDateTime.parse("2020-09-09T18:00:00").atZone(ZoneId.of("UTC")).toInstant().toEpochMilli();
+            setup(sourceIndex, 1, 0, startTime);
+            final DownsampleConfig config = new DownsampleConfig(randomInterval());
+            final DownsampleActionSingleNodeTests.SourceSupplier sourceSupplier = () -> {
+                final String ts = randomDateForInterval(config.getInterval(), startTime);
+                double counterValue = DATE_FORMATTER.parseMillis(ts);
+                final List<String> dimensionValues = new ArrayList<>(5);
+                for (int j = 0; j < randomIntBetween(1, 5); j++) {
+                    dimensionValues.add(randomAlphaOfLength(6));
+                }
+                return XContentFactory.jsonBuilder()
+                    .startObject()
+                    .field(FIELD_TIMESTAMP, ts)
+                    .field(FIELD_DIMENSION_1, randomFrom(dimensionValues))
+                    .field(FIELD_DIMENSION_2, randomIntBetween(1, 10))
+                    .field(FIELD_METRIC_COUNTER, counterValue)
+                    .endObject();
+            };
+            int indexedDocs = bulkIndex(sourceIndex, sourceSupplier, DOC_COUNT);
+            prepareSourceIndex(sourceIndex);
+            final CountDownLatch disruptionStart = new CountDownLatch(1);
+            final CountDownLatch disruptionEnd = new CountDownLatch(1);
+
+            new Thread(new Disruptor(cluster, sourceIndex, new DisruptionListener() {
+                @Override
+                public void disruptionStart() {
+                    disruptionStart.countDown();
+                }
+
+                @Override
+                public void disruptionEnd() {
+                    disruptionEnd.countDown();
+                }
+            }, masterNodes.get(0), (ignored) -> {
+                try {
+                    cluster.rollingRestart(new InternalTestCluster.RestartCallback() {
+                        @Override
+                        public boolean validateClusterForming() {
+                            return true;
+                        }
+                    });
+                } catch (Exception e) {
+                    throw new RuntimeException(e);
+                }
+            })).start();
+
+            if (startRollupTask(sourceIndex, rollupIndex, config, disruptionStart, disruptionEnd) == false) {
+                return;
+            }
+            waitUntil(() -> cluster.client().admin().cluster().preparePendingClusterTasks().get().pendingTasks().isEmpty());
+            ensureStableCluster(cluster.numDataAndMasterNodes());
+            assertRollupIndex(cluster, sourceIndex, rollupIndex, indexedDocs);
+        }
+    }
+
+    private boolean startRollupTask(
```

Comment:
Would it make sense to document this method? 
It's not immediately apparent what this method does and what the boolean it returns represents (i.e. it does more than just starting a task)

Would `maybeStartRollupTaskDuringDisruption` be a more accurate name?